misc/pytorch_toolkit/breast_ultrasound_simulation/src/solve.py

```diff
@@ -0,0 +1,335 @@
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.utils.tensorboard import SummaryWriter
+import torchvision
+import os
+import shutil
+import numpy as np
+import time
+from model import GeneratorModel, DiscriminatorModel, GeneratorInter, Generator3dInter
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+print(torch.cuda.is_available())
+
+
+class solver():
+    def __init__(self, args, train_data=None, test_data=None, restore=0):
+        self.args = args
+        self.train_data = train_data
+        self.test_data = test_data
+
+        self.gen = GeneratorModel(1).to(device)
+        self.dis = DiscriminatorModel(2).to(device)
+
+        self.recon_l = nn.L1Loss().to(device)
+        self.adv_l = nn.BCELoss().to(device)
+
+        self.g_opt = optim.Adam(self.gen.parameters(), lr=args.lr)
+        self.d_opt = optim.Adam(self.dis.parameters(), lr=args.lr)
+
+        self.output_dir = os.path.join('../infer_results', args.name)
+        if not os.path.isdir(self.output_dir):
+            os.mkdir(self.output_dir)
+
+        self.log_dir = '../logs/' + args.name
+        self.writer = SummaryWriter(self.log_dir)
+        if not os.path.isdir("../checkpoints/" + args.name):
+            os.mkdir("../checkpoints/" + args.name)
+
+        self.log = args.log_step != -1
+        print("Log Dir: ../logs")
+        print("Checkpoint Dir: ../checkpoints")
+
+        if restore or args.test:
+            state_dict = torch.load(
+                os.path.join(
+                    "../checkpoints",
+                    args.model_name))
+            self.gen.load_state_dict(state_dict["generator1_weights"])
+            self.dis.load_state_dict(state_dict["discriminator1_weights"])
+            self.g_opt.load_state_dict(state_dict["generator1_optimizer"])
+            self.d_opt.load_state_dict(state_dict["discriminator1_optimizer"])
+            print("restored")
+
+    def set_requires_grad(self, net, requires_grad=False):
+        """Set :requies_grad=Fasle for all the networks to avoid unnecessary computations
```

Comment:
```suggestion
        """Set :requires_grad=Fasle for all the networks to avoid unnecessary computations
```

By the way, it is strange to see docstring for only some functions. Add docstring to other functions and classes or remove them at all.